pkg/clusterapi/proxy.go

```diff
@@ -8,12 +8,64 @@ import (
 	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
 
 	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/logger"
 	"github.com/aws/eks-anywhere/pkg/templater"
 )
 
 //go:embed config/http-proxy.conf
 var proxyConfig string
 
+func proxy(clusterSpec *cluster.Spec) bootstrapv1.ProxyConfiguration {
+	return bootstrapv1.ProxyConfiguration{
+		HTTPSProxy: clusterSpec.Cluster.Spec.ProxyConfiguration.HttpsProxy,
+		NoProxy:    noProxyList(clusterSpec.Cluster.Spec),
+	}
+}
+
+// SetProxyConfigInKubeadmControlPlane sets up proxy configuration in kubeadmControlPlane.
+// Currently only support proxy setting for bottlerocket and ubuntu. No-op for any other OS.
+func SetProxyConfigInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane, clusterSpec *cluster.Spec, os v1alpha1.OSFamily) error {
+	if clusterSpec.Cluster.Spec.ProxyConfiguration == nil {
+		return nil
+	}
+
+	switch os {
```

Comment:
I would prefer is this check was made at the provider level and we had two methods: SetProxyConfigInKubeadmControlPlaneForBottlerocket` and `SetProxyConfigInKubeadmControlPlaneForUbuntu`.

The os support is still a provider concern IMO, at least from the eks-a perspective.